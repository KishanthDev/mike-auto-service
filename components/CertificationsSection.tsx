"use client";

import React from "react";
import type { Certification } from "@/types/data";

interface CertificateSectionProps {
  certifications: Certification[];
}

export default function CertificationsSection({certifications}:CertificateSectionProps) {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white">
        Certifications & Expertise
      </h2>

      <p className="text-sm md:text-base text-gray-700 dark:text-gray-300 mt-2">
        Our technicians are specially trained and certified to work on all major
        vehicle makes and models.
      </p>

      <div className="flex flex-wrap gap-4 mt-4 md:mt-6 justify-center sm:justify-start">
        {certifications.map((badge, idx) => (
          <div
            key={idx}
            className="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full 
                       bg-gray-100 dark:bg-gray-800 border dark:border-gray-600 
                       text-sm md:text-base font-semibold text-gray-900 dark:text-white"
          >
            {badge}
          </div>
        ))}
      </div>
    </section>
  );
}
