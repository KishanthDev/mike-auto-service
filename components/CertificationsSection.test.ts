import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import CertificationsSection from "./CertificationsSection";
import type { Certification } from "@/types/data";

describe("CertificationsSection", () => {
  const mockCertifications: Certification[] = [
    "ASE Certified",
    "Ford Master Tech",
    "Toyota Specialist",
    "Hyundai Certified",
    "GM World Class",
  ];

  it("renders the section with correct heading", () => {
    render(
      React.createElement(CertificationsSection, {
        certifications: mockCertifications,
      }),
    );

    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Certifications & Expertise");
    expect(heading).toHaveClass(
      "text-2xl",
      "md:text-3xl",
      "font-bold",
      "border-b-2",
      "border-red-600",
      "pb-2",
      "text-[#1a3c6e]",
      "dark:text-white",
    );
  });

  it("renders the description text", () => {
    render(
      React.createElement(CertificationsSection, {
        certifications: mockCertifications,
      }),
    );

    const description = screen.getByText(
      /Our technicians are specially trained and certified to work on all major vehicle makes and models/i,
    );
    expect(description).toBeInTheDocument();
    expect(description).toHaveClass(
      "text-sm",
      "md:text-base",
      "text-gray-700",
      "dark:text-gray-300",
      "mt-2",
    );
  });

  it("handles empty certifications array", () => {
    render(React.createElement(CertificationsSection, { certifications: [] }));

    // Heading and description should still render
    expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument();
    expect(
      screen.getByText(/Our technicians are specially trained/i),
    ).toBeInTheDocument();

    // No badges should be rendered
    expect(screen.queryAllByTestId("certification-badge")).toHaveLength(0);
  });

  it("matches snapshot with certifications", () => {
    const { asFragment } = render(
      React.createElement(CertificationsSection, {
        certifications: mockCertifications,
      }),
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with empty certifications", () => {
    const { asFragment } = render(
      React.createElement(CertificationsSection, { certifications: [] }),
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
