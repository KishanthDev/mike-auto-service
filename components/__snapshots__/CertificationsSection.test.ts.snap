// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificationsSection matches snapshot with certifications 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Certifications & Expertise
    </h2>
    <p
      class="text-sm md:text-base text-gray-700 dark:text-gray-300 mt-2"
    >
      Our technicians are specially trained and certified to work on all major vehicle makes and models.
    </p>
    <div
      class="flex flex-wrap gap-4 mt-4 md:mt-6 justify-center sm:justify-start"
    >
      <div
        class="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full  bg-gray-100 dark:bg-gray-800 border dark:border-gray-600  text-sm md:text-base font-semibold text-gray-900 dark:text-white"
      >
        ASE Certified
      </div>
      <div
        class="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full  bg-gray-100 dark:bg-gray-800 border dark:border-gray-600  text-sm md:text-base font-semibold text-gray-900 dark:text-white"
      >
        Ford Master Tech
      </div>
      <div
        class="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full  bg-gray-100 dark:bg-gray-800 border dark:border-gray-600  text-sm md:text-base font-semibold text-gray-900 dark:text-white"
      >
        Toyota Specialist
      </div>
      <div
        class="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full  bg-gray-100 dark:bg-gray-800 border dark:border-gray-600  text-sm md:text-base font-semibold text-gray-900 dark:text-white"
      >
        Hyundai Certified
      </div>
      <div
        class="w-16 h-16 md:w-20 md:h-20 flex justify-center items-center rounded-full  bg-gray-100 dark:bg-gray-800 border dark:border-gray-600  text-sm md:text-base font-semibold text-gray-900 dark:text-white"
      >
        GM World Class
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`CertificationsSection matches snapshot with empty certifications 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Certifications & Expertise
    </h2>
    <p
      class="text-sm md:text-base text-gray-700 dark:text-gray-300 mt-2"
    >
      Our technicians are specially trained and certified to work on all major vehicle makes and models.
    </p>
    <div
      class="flex flex-wrap gap-4 mt-4 md:mt-6 justify-center sm:justify-start"
    />
  </section>
</DocumentFragment>
`;
